rxjava/src/main/java/com/linecorp/armeria/common/rxjava/RequestContextAssembly.java

```diff
@@ -0,0 +1,231 @@
+        RxJavaPlugins.setOnMaybeAssembly(
+                compose(oldOnMaybeAssembly, new ConditionalOnCurrentRequestContextFunction<Maybe>() {
+                    @Override
+                    Maybe applyActual(Maybe m, RequestContext ctx) {
+                        if (!(m instanceof Callable)) {
+                            return new RequestContextMaybe(m, ctx);
+                        }
+                        if (m instanceof ScalarCallable) {
+                            return new RequestContextScalarCallableMaybe(m, ctx);
+                        }
+                        return new RequestContextCallableMaybe(m, ctx);
+                    }
+                }));
+
+        oldOnFlowableAssembly = RxJavaPlugins.getOnFlowableAssembly();
+        RxJavaPlugins.setOnFlowableAssembly(
+                compose(oldOnFlowableAssembly, new ConditionalOnCurrentRequestContextFunction<Flowable>() {
+                    @Override
+                    Flowable applyActual(Flowable f, RequestContext ctx) {
+                        if (!(f instanceof Callable)) {
+                            return new RequestContextFlowable(f, ctx);
+                        }
+                        if (f instanceof ScalarCallable) {
+                            return new RequestContextScalarCallableFlowable(f, ctx);
+                        }
+                        return new RequestContextCallableFlowable(f, ctx);
+                    }
+                }));
+
+        oldOnConnectableFlowableAssembly = RxJavaPlugins.getOnConnectableFlowableAssembly();
+        RxJavaPlugins.setOnConnectableFlowableAssembly(
+                compose(oldOnConnectableFlowableAssembly,
+                        new ConditionalOnCurrentRequestContextFunction<ConnectableFlowable>() {
+                            @Override
+                            ConnectableFlowable applyActual(
+                                    ConnectableFlowable cf,
+                                    RequestContext ctx) {
+                                return new RequestContextConnectableFlowable(
+                                        cf, ctx);
+                            }
+                        }
+                ));
+
+        oldOnParallelAssembly = RxJavaPlugins.getOnParallelAssembly();
+        RxJavaPlugins.setOnParallelAssembly(
+                compose(oldOnParallelAssembly,
+                        new ConditionalOnCurrentRequestContextFunction<ParallelFlowable>() {
+                            @Override
+                            ParallelFlowable applyActual(ParallelFlowable pf, RequestContext ctx) {
+                                return new RequestContextParallelFlowable(pf, ctx);
+                            }
+                        }
+                ));
+    }
+
+    /**
+     * Disable {@link RequestContext} during operators.
+     */
+    public static synchronized void disable() {
+        RxJavaPlugins.setOnObservableAssembly(oldOnObservableAssembly);
```

Comment:
- Set `null` to the `old*` once disabled?
- Make `disable()` no-op if disabled already or a user did not enable yet.